test/job/concurrentreads/concurrentreads.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2021 Andy Dalton
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/adalton/teleport-exercise/pkg/jobmanager"
+)
+
+func runTest() {
+
+	job := jobmanager.NewJob("theOwner", "my-test", nil,
+		"/bin/bash",
+		"-c",
+		"for ((i = 0; i < 100; ++i)); do for((j = 0; j < 1000; ++j)); do echo $RANDOM; done; sleep 0.25; done",
```

Comment:
Suggestion: print numbers in order, then make sure the goroutines see the expected output.

Reply:
I've enhanced the test to ensure all goroutines see the same values, the same number of values, and in the same order.  It'll panic on failure and terminate successfully on success.